src/epcpm/cli/sunspectostaticmodbus.py

```diff
@@ -117,14 +138,40 @@ def create_function_data_bitfield(
     else:
         function_data_bitfield.type_uuid = None
     function_data_bitfield.size = int(input_sunspec_data["size"])
-    # A bitfield would never be a scale factor, so no need to check for uuid in scale_factor_uuid_map.
+    # A bitfield is never a scale factor, so no need to check for uuid in scale_factor_uuid_map.
     function_data_bitfield.uuid = generate_uuid()
     function_data_bitfield.address = int(input_sunspec_data["modbus_address"])
 
     return function_data_bitfield
 
 
+def create_function_data_bitfield_member(
```

Comment:
There's a fair bit of reused code. Is it worthwhile to synthesize the data_bitfield and data_bitfield_member functions like so?:

```
def create_function_data_bitfield(input_sunspec_data, sunspec_types, staticmodbus_types, member=False):
    if member:
        function_data_bitfield = epcpm.staticmodbusmodel.FunctionDataBitfieldMember()
    else:
        function_data_bitfield = epcpm.staticmodbusmodel.FunctionDataBitfield()
    function_data_bitfield.parameter_uuid = (
        input_sunspec_data["parameter_uuid"]
        if input_sunspec_data["parameter_uuid"]
        else None
    )
    if input_sunspec_data["type_uuid"]:
        function_data_bitfield.type_uuid = map_type_uuid(
            input_sunspec_data["type_uuid"], sunspec_types, staticmodbus_types
        )
    else:
        function_data_bitfield.type_uuid = None
    if not member:
        function_data_bitfield.size = int(input_sunspec_data["size"])
        function_data_bitfield.address = int(input_sunspec_data["modbus_address"])
    else:
        function_data_bitfield.bit_offset = int(input_sunspec_data["bit_offset"])
        function_data_bitfield.bit_length = int(input_sunspec_data["bit_length"])
    function_data_bitfield.uuid = generate_uuid()
    return function_data_bitfield
```

Reply:
Refactored the three `create_function_data_*` methods in 73afd44b3cf2d463467b95ab4664543c5ee732e7